### ISSUE TITLE
Corrupted serial output

### ISSUE THREAD

**Issue by dmikushin** [hexaedron/PostNeoSWSerial#1]
Hi @hexaedron , thanks for moving on with NeoSWSerial!

With the following code, I get garbage in minicom, screenshot attached.

![Screenshot from 2023-10-24 10-15-33](https://github.com/hexaedron/PostNeoSWSerial/assets/5672874/0d79626e-c48e-44d2-9e23-00bc0fad6499)

SoftwareSerial works normally for the same sketch. My LGT8F328P Nano is running at 32MHz, in `PostNeoSWSerial.cpp` I noticed:

```c
#if (F_CPU == 16000000L) || defined(lgtf8x)
```

So I had to define `lgtf8x` macro in the board compiler flags. Am I supposed to do this, or should I instead switch to 16MHz? If yes, I don't know how to change `Clock Source` to 16MHz, because it is marked "External" and does not work at all.

```c
//#include <SoftwareSerial.h>
#include <PostNeoSWSerial.h>

const uint8_t RX = 9;
const uint8_t TX = 10;

//SoftwareSerial swSerial(RX, TX);
PostNeoSWSerial swSerial(RX, TX);

void setup() {
  Serial.begin(115200);

  while (!Serial)
  {
    // Wait for serial port to connect. Needed for native USB port only
    delay(100);
  }

  // Define pin modes for TX and RX
  pinMode(RX, INPUT);
  pinMode(TX, OUTPUT);
  swSerial.begin(9600);
}

void loop() { // run over and over
  Serial.println("tick");
  swSerial.println("tick");
  /*while (swSerial.available()) {
    Serial.write(swSerial.read());
  }*/
  while (Serial.available()) {
    swSerial.write(static_cast<char>(Serial.read()));
  }
  delay(100);
}
```

**Comment by dmikushin** [comment]
Important to note that my board is green Nano:

![lgt8f328p_nano_qfp32l_pinout-1-1024x627](https://github.com/hexaedron/PostNeoSWSerial/assets/5672874/dbbd0e54-9f13-4ca0-92fd-68137cab1634)

**Comment by dmikushin** [comment]
Since my board has no external clock, I can't switch to 16 MHz clock. But I'm able to set clock divider to 2, and this way IT WORKS, the described issue goes away.

![Screenshot from 2023-10-24 11-30-31](https://github.com/hexaedron/PostNeoSWSerial/assets/5672874/c0845f06-6873-47e4-a1c3-02c5697457fe)

**Comment by hexaedron** [comment]
Clock divider to 2 is a workaround, i am working on this issue now.

**Comment by hexaedron** [comment]
Fixed in 1.0.3 so you can enjoy your LGT8F328P at 32 MHz without a clock divider!
